llpc/tool/amdllpc.cpp

```diff
@@ -666,16 +675,59 @@ static Error processInputs(ICompiler *compiler, InputSpecGroup &inputSpecs) {
     compileInfo.autoLayoutDesc = false;
     if (Error err = processInputPipeline(compiler, compileInfo, firstInput, Unlinked, IgnoreColorAttachmentFormats))
       return err;
+
+    if (isRayTracingPipeline(compileInfo.stageMask)) {
+      if (LlpcRaytracingModeSetting.getNumOccurrences())
+        compileInfo.rayTracePipelineInfo.mode = LlpcRaytracingModeSetting;
+    }
   } else {
     compileInfo.autoLayoutDesc = true;
     if (Error err = processInputStages(compileInfo, inputSpecs, ValidateSpirv, NumThreads))
       return err;
 
-    compileInfo.pipelineType =
-        isComputePipeline(compileInfo.stageMask) ? VfxPipelineTypeCompute : VfxPipelineTypeGraphics;
-
-    if (isRayTracingPipeline(compileInfo.stageMask))
+    if (isRayTracingPipeline(compileInfo.stageMask)) {
       compileInfo.pipelineType = VfxPipelineTypeRayTracing;
+      compileInfo.rayTracePipelineInfo.indirectStageMask = 0xFFFFFFFF;
+      compileInfo.rayTracePipelineInfo.pipelineLibStageMask = 0xFFFFFFFF;
+      compileInfo.rayTracePipelineInfo.hasPipelineLibrary = true;
+
+      standaloneRtShaders.resize(compileInfo.shaderModuleDatas.size());
+      memset(&standaloneRtShaders[0], 0, sizeof(PipelineShaderInfo) * standaloneRtShaders.size());
+      compileInfo.rayTracePipelineInfo.pShaders = &standaloneRtShaders[0];
+
+      compileInfo.unlinked = true;
+      compileInfo.doAutoLayout = true;
+      compileInfo.autoLayoutDesc = true;
+
+      compileInfo.rayTracePipelineInfo.mode = LlpcRaytracingModeSetting;
+      compileInfo.rayTracePipelineInfo.maxRecursionDepth = 1;
+
+      RtState &state = compileInfo.rayTracePipelineInfo.rtState;
+      state.pipelineFlags = 0;
```

Comment:
I'd prefer to put the default initializer in a separate method, since it clutters the code here.

Reply:
One can go either way on this.